src/models/curriculum.ts

```diff
@@ -11,3 +13,34 @@ export const CurriculumModel = types
   });
 
 export type CurriculumModelType = typeof CurriculumModel.Type;
+
+/*
+  createFromJson
+
+  The JSON representation contains strings for things like SectionTypes.
+  CurriculumModel.create() expects proper TypeScript enumerated values instead.
+  This function recursively replaces SectionType strings with the corresponding
+  SectionType enumerated values and returns the resulting CurriculumModel.
+ */
+export function createFromJson(json: any) {
+  const snapshot = replaceSectionTypes(json);
+  return CurriculumModel.create(snapshot);
+}
+
+/*
+  replaceSectionTypes
+
+  Recursively replaces SectionType strings in 'type' fields with corresponding
+  SectionType enumerated values.
+ */
+function replaceSectionTypes(obj: {}): SnapshotIn<typeof CurriculumModel> {
```

Comment:
I guess eventually we'll have more generic functions for replacing other non-strings, like additional `enum`s or images?

Reply:
Potentially. I figured I'd start with a tailored local function and we can see how often the need arises.